internal/querynodev2/handlers.go

```diff
@@ -249,6 +253,66 @@ func (node *QueryNode) querySegments(ctx context.Context, req *querypb.QueryRequ
 	}, nil
 }
 
+func (node *QueryNode) optimizeSearchParams(ctx context.Context, req *querypb.SearchRequest, delegator delegator.ShardDelegator) (*querypb.SearchRequest, error) {
+	// no hook applied, just return
+	if node.queryHook == nil {
+		return req, nil
+	}
+
+	log := log.Ctx(ctx)
+
+	serializedPlan := req.GetReq().GetSerializedExprPlan()
+	// plan not found
+	if serializedPlan == nil {
+		log.Warn("serialized plan not found")
+		return req, merr.WrapErrParameterInvalid("serialized search plan", "nil")
+	}
+
+	channelNum := req.GetTotalChannelNum()
+	// not set, change to conservative channel num 1
+	if channelNum <= 0 {
+		channelNum = 1
+	}
+
+	plan := planpb.PlanNode{}
+	err := proto.Unmarshal(serializedPlan, &plan)
+	if err != nil {
+		log.Warn("failed to unmarshal plan", zap.Error(err))
+		return nil, merr.WrapErrParameterInvalid("valid serialized search plan", "no unmarshalable one", err.Error())
+	}
+
+	switch plan.GetNode().(type) {
+	case *planpb.PlanNode_VectorAnns:
+		sealed, _, version := delegator.GetDistribution().GetCurrent()
+		delegator.GetDistribution().FinishUsage(version)
+		// use shardNum * segments num in shard to estimate total segment number
+		estSegmentNum := len(sealed) * int(channelNum)
```

Comment:
It's hard to recognize what "est" means, is it a common and widely used abbr?

Reply:
est. is short for estimated